Tailviewer.Test/Ui/Controls/LogViewerControlTest.cs

```diff
@@ -1,529 +1,544 @@
-			dataSource.SetupProperty(x => x.DisplayMode);
-
-			_control.DataSource = dataSource.Object;
-
-			_control.MergedDataSourceDisplayMode = DataSourceDisplayMode.CharacterCode;
-			dataSource.Object.DisplayMode.Should().Be(DataSourceDisplayMode.CharacterCode);
-
-			_control.MergedDataSourceDisplayMode = DataSourceDisplayMode.Filename;
-			dataSource.Object.DisplayMode.Should().Be(DataSourceDisplayMode.Filename);
-		}
-
-		[Test]
-		[Description("Verifies that changes to the MergedDataSourceDisplayMode property are ignored if the view model isn't a merged one")]
-		public void TestChangeMergedDataSourceDisplayMode2()
-		{
-			var dataSource = new Mock<IDataSourceViewModel>();
-
-			_control.DataSource = dataSource.Object;
-
-			new Action(() => _control.MergedDataSourceDisplayMode = DataSourceDisplayMode.CharacterCode).ShouldNotThrow();
-			new Action(() => _control.MergedDataSourceDisplayMode = DataSourceDisplayMode.Filename).ShouldNotThrow();
-		}
-
-		[Test]
-		[Description("Verifies that the display mode of the new data source is used")]
-		public void TestChangeDataSource([Values(DataSourceDisplayMode.Filename, DataSourceDisplayMode.CharacterCode)] DataSourceDisplayMode displayMode)
-		{
-			var dataSource = new Mock<IMergedDataSourceViewModel>();
-			dataSource.SetupProperty(x => x.DisplayMode);
-			dataSource.Object.DisplayMode = displayMode;
-
-			_control.DataSource = dataSource.Object;
-			_control.MergedDataSourceDisplayMode.Should().Be(displayMode, "because the view model determines the initial display mode and thus the control should just use that");
-			dataSource.Object.DisplayMode.Should().Be(displayMode, "because the display mode of the view model shouldn't have been changed in the process");
-		}
-
-		[Test]
-		[Description("Verifies that the settings are simply forwarded to the LogEntryListView")]
-		public void TestChangeSettings()
-		{
-			var settings = new LogViewerSettings();
-
-			_control.Settings = settings;
-			_control.PART_ListView.Settings.Should().BeSameAs(settings);
-		}
-	}
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using FluentAssertions;
+using Moq;
+using NUnit.Framework;
+using Tailviewer.BusinessLogic;
+using Tailviewer.BusinessLogic.ActionCenter;
+using Tailviewer.BusinessLogic.DataSources;
+using Tailviewer.BusinessLogic.LogFiles;
+using Tailviewer.Settings;
+using Tailviewer.Ui.Controls.LogView;
+using Tailviewer.Ui.ViewModels;
+
+namespace Tailviewer.Test.Ui.Controls
+{
+	[TestFixture]
+	[Apartment(ApartmentState.STA)]
+	public sealed class LogViewerControlTest
+	{
+		private Mock<IActionCenter> _actionCenter;
+		private LogViewerControl _control;
+		private SingleDataSourceViewModel _dataSource;
+		private ILogFileFactory _logFileFactory;
+		private ManualTaskScheduler _scheduler;
+		private Mock<IApplicationSettings> _settings;
+
+		[OneTimeSetUp]
+		public void OneTimeSetUp()
+		{
+			_scheduler = new ManualTaskScheduler();

```

Comment:
Please don't re-use scheduler between tests, they're created in setup for a reason: This way individual tests don't have to perform a full cleanup

Reply:
Just noticed that you didn't change this, can you move the construction nevertheless? I guess I was lazy

Reply:
I guess you did it because of the next line. PluginLogFileFactory needs a scheduler as an argument.